### ISSUE TITLE
[Bug]: OG Heists issues when "Spoof Player Model" is enabled

### ISSUE THREAD

**Issue by GTMoraes** [Yimura/YimMenu#1920]
### Describe the bug

Fleeca job: It starts, but the last mission, where you must drive the Armored Kuruma to the bank, if I (yimmenu user) get into the car before the driver, the driver is unable to drive until I leave the car. After he moves the car a little bit, it goes through and mission can be finished.

Other heists:
So far, I couldn't get past the drawing board phase. I only joined already active heists, and never hosted one.
Either it's the initial drawing board where you're supposed to select one of the preps, or the final drawing board where cuts and roles are selected.

In either case, it's like everybody's AFK, nobody readys up, nobody changes roles, and the "Continue" timer on the bottom right corner is 0:00 since the beginning. Nobody answers the chat either.

I've tried disabling protections, reactions and whatnot. Restarting the game without YimMenu, the heists are possible to start 

### Steps To Reproduce

What I do:

1. Join a heist (On a Call)
2. Wait until lobby fills up
3. If Fleeca Job, it goes through. Just don't enter the Kuruma on the last mission before the driver.
4. If any other mission, whenever it gets on the drawing board, it's like everybody's AFK and not responding.

### Expected Behavior

Play the heists.

### Logs and/or screenshots

```
names edited to prevent identification

[00:28:00.2501169][INFO/main.cpp:65] Yim's Menu Initializing
[00:28:00.2501412][INFO/main.cpp:66] Git Info
	Branch:	master
	Hash:	c2d91c11aa5ae86db596cbaa9d50754332e870d3
	Date:	Fri Aug 4 23:19:39 2023
[00:28:00.2501551][DEBUG/thread_pool.cpp:21] Resizing thread pool from 0 to 4
[00:28:00.2508249][INFO/main.cpp:69] Thread pool initialized.
[00:28:00.2588553][INFO/settings.cpp:79] Updating settings.
[00:28:00.2613345][INFO/main.cpp:72] Settings Loaded.
[00:28:00.2613420][DEBUG/thread_pool.cpp:95] Thread 24504 executing settings.cpp:18
[00:28:00.2647767][INFO/pointers.hpp:94] GTA5 pointers cache is up to date, using it.
[00:28:00.2647975][INFO/pointers.cpp:1548] Loading pointers instance from cache
[00:28:00.2647990][INFO/pointers.cpp:1557] Pointers cache: Loading 181 fields from the cache
[00:28:00.2648084][DEBUG/module.cpp:50] Waiting for socialclub.dll...
[00:28:00.2682610][INFO/pointers.hpp:94] Social Club pointers cache is up to date, using it.
[00:28:00.2682712][INFO/pointers.cpp:1548] Loading pointers instance from cache
[00:28:00.2682731][INFO/pointers.cpp:1557] Pointers cache: Loading 5 fields from the cache
[00:28:00.2683034][INFO/main.cpp:75] Pointers initialized.
[00:28:00.2684127][INFO/main.cpp:78] Byte Patch Manager initialized.
[00:28:00.7679668][INFO/main.cpp:81] Renderer initialized.
[00:28:00.9810132][INFO/main.cpp:85] Fiber pool initialized.
[00:28:01.4866894][INFO/main.cpp:88] Translation Service initialized.
[00:28:01.5060900][INFO/main.cpp:91] Hooking initialized.
[00:28:01.5233490][DEBUG/gta_data_service.cpp:162] Loading data from cache.
[00:28:01.5233580][INFO/gta_data_service.cpp:174] Loading 1029 peds from cache.
[00:28:01.5236546][INFO/gta_data_service.cpp:192] Loading 815 vehicles from cache.
[00:28:01.5238805][INFO/gta_data_service.cpp:209] Loading 117 weapons from cache.
[00:28:01.5238851][INFO/gta_data_service.cpp:210] Loading 378 weapon components from cache.
[00:28:01.5238950][DEBUG/gta_data_service.cpp:168] Loaded all data from cache.
[00:28:01.6338651][INFO/tunables_service.cpp:21] Loading tunables from cache
[00:28:01.6339001][INFO/main.cpp:111] Registered service instances...
[00:28:01.6338948][DEBUG/thread_pool.cpp:95] Thread 22856 executing tunables_service.cpp:24
[00:28:01.6340089][INFO/main.cpp:132] Scripts registered.
[00:28:01.6905304][INFO/main.cpp:135] Hooking enabled.
[00:28:01.6921547][INFO/main.cpp:138] Dynamic native hooker initialized.
[00:28:01.7013360][INFO/lua_module.cpp:226] Loaded CasinoPacino.lua
[00:28:01.7013987][INFO/main.cpp:141] Lua manager initialized.
[00:28:01.7014197][DEBUG/thread_pool.cpp:95] Thread 28164 executing player_database_service.cpp:276
[00:28:01.7781003][INFO/backend.cpp:51] Starting script: Self
[00:28:01.7781106][INFO/backend.cpp:76] Starting script: Weapons
[00:28:01.7781236][INFO/backend.cpp:99] Starting script: Vehicles
[00:28:01.7781274][INFO/backend.cpp:115] Starting script: Miscellaneous
[00:28:01.7781337][INFO/backend.cpp:135] Starting script: Remote
[00:28:01.7781469][INFO/backend.cpp:150] Starting script: Ls customs
[00:28:01.7781488][INFO/backend.cpp:162] Starting script: Rainbow paint
[00:28:01.7781509][INFO/backend.cpp:174] Starting script: Disable Control Action
[00:28:01.7781531][INFO/backend.cpp:187] Starting script: World
[00:28:45.6324523][DEBUG/thread_pool.cpp:95] Thread 22856 executing translation_service.cpp:95
[00:28:48.4420788][DEBUG/thread_pool.cpp:95] Thread 22856 executing view_settings.cpp:54
[00:29:04.3022046][DEBUG/thread_pool.cpp:95] Thread 22856 executing view_settings.cpp:54
[00:29:08.5294258][DEBUG/thread_pool.cpp:95] Thread 22856 executing translation_service.cpp:95
[00:29:08.5309556][INFO/translation_service.cpp:158] Translations for 'zh_TW' does not exist, downloading from https://raw.githubusercontent.com/GTMoraes/Translations/master
[00:29:11.9000006][DEBUG/thread_pool.cpp:95] Thread 22856 executing translation_service.cpp:95
[00:29:11.9016064][INFO/translation_service.cpp:158] Translations for 'zh_CN' does not exist, downloading from https://raw.githubusercontent.com/GTMoraes/Translations/master
[00:29:14.5079745][DEBUG/thread_pool.cpp:95] Thread 22856 executing translation_service.cpp:95
[00:29:17.8066681][DEBUG/thread_pool.cpp:95] Thread 22856 executing translation_service.cpp:95
[00:29:33.8253268][DEBUG/thread_pool.cpp:95] Thread 22856 executing translation_service.cpp:95
[00:30:43.0581950][DEBUG/thread_pool.cpp:95] Thread 22856 executing view_settings.cpp:54
[00:30:45.5439098][DEBUG/thread_pool.cpp:95] Thread 22856 executing translation_service.cpp:95
[00:30:47.1600231][DEBUG/thread_pool.cpp:95] Thread 22856 executing translation_service.cpp:95
[00:30:58.5187634][DEBUG/thread_pool.cpp:95] Thread 22856 executing translation_service.cpp:95
[00:30:59.7997505][DEBUG/thread_pool.cpp:95] Thread 22856 executing translation_service.cpp:95
[00:34:16.4233749][WARN/animations.hpp:94] Animations file is not in directory. https://raw.githubusercontent.com/DurtyFree/gta-v-data-dumps/master/animDictsCompact.json
[00:38:33.3090963][WARN/animations.hpp:94] Animations file is not in directory. https://raw.githubusercontent.com/DurtyFree/gta-v-data-dumps/master/animDictsCompact.json
[00:41:12.8787511][WARN/animations.hpp:94] Animations file is not in directory. https://raw.githubusercontent.com/DurtyFree/gta-v-data-dumps/master/animDictsCompact.json
[00:41:24.2105554][DEBUG/thread_pool.cpp:95] Thread 26476 executing animations.hpp:103
[00:42:03.3477915][WARN/reaction.cpp:25] Received Modder Detection from XXXXXXXXX (000000000000)
[00:42:29.9564606][WARN/animations.hpp:87] Anim list already contained data, overwriting...
[00:42:29.9913674][DEBUG/thread_pool.cpp:95] Thread 26476 executing animations.hpp:103
[00:52:51.6518062][WARN/reaction.cpp:25] Received Send To Interior from XXXXXXXXX (000000000000)
[01:04:21.6028611][DEBUG/thread_pool.cpp:95] Thread 22856 executing translation_service.cpp:95
[01:04:34.4318819][WARN/reaction.cpp:25] Received Send To Interior from XXXXXXXXX (000000000000)
[01:12:50.7689591][WARN/reaction.cpp:25] Received Send To Interior from XXXXXXXXX (000000000000)
[01:41:08.8961809][DEBUG/script_function.cpp:35] Found pattern RSD in script PauseMenu_Multiplayer
[02:01:35.5429652][INFO/main.cpp:152] Scripts unregistered.
[02:01:35.5453398][INFO/main.cpp:155] Lua manager uninitialized.
[02:01:35.6055284][INFO/main.cpp:158] Hooking disabled.
[02:01:35.6057183][INFO/main.cpp:161] Dynamic native hooker uninitialized.
[02:01:35.6057779][DEBUG/thread_pool.cpp:108] Thread 22856 exiting...
[02:01:35.6057847][DEBUG/thread_pool.cpp:108] Thread 26476 exiting...
[02:01:35.6812119][DEBUG/thread_pool.cpp:108] Thread 24504 exiting...
[02:01:36.3448475][DEBUG/thread_pool.cpp:108] Thread 28164 exiting...
[02:01:36.3453526][INFO/main.cpp:169] Destroyed thread pool.
[02:01:36.3453630][INFO/main.cpp:172] Script Connection Service reset.
[02:01:36.3467730][INFO/main.cpp:174] Tunables Service reset.
[02:01:36.3467828][INFO/main.cpp:176] Hotkey Service reset.
[02:01:36.3467860][INFO/main.cpp:178] Matchmaking Service reset.
[02:01:36.3469361][INFO/main.cpp:180] Player Database Service reset.
[02:01:36.3469413][INFO/main.cpp:182] API Service reset.
[02:01:36.3563455][INFO/main.cpp:184] Script Patcher Service reset.
[02:01:36.3563720][INFO/main.cpp:186] Gui Service reset.
[02:01:36.3578028][INFO/main.cpp:188] GTA Data Service reset.
[02:01:36.3578257][INFO/main.cpp:190] Vehicle Service reset.
[02:01:36.3578283][INFO/main.cpp:192] Model Preview Service reset.
[02:01:36.3578300][INFO/main.cpp:194] Mobile Service reset.
[02:01:36.3578368][INFO/main.cpp:196] Player Service reset.
[02:01:36.3578478][INFO/main.cpp:198] Pickup Service reset.
[02:01:36.3578513][INFO/main.cpp:200] Custom Text Service reset.
[02:01:36.3578871][INFO/main.cpp:202] Context Service reset.
[02:01:36.3578905][INFO/main.cpp:204] Xml Vehicles Service reset.
[02:01:36.3578921][INFO/main.cpp:205] Services uninitialized.
[02:01:36.3579245][INFO/main.cpp:208] Hooking uninitialized.
[02:01:36.3579323][INFO/main.cpp:211] Fiber pool uninitialized.
[02:01:36.3682447][INFO/main.cpp:214] Renderer uninitialized.
[02:01:36.3683798][INFO/main.cpp:217] Byte Patch Manager uninitialized.
[02:01:36.3683860][INFO/main.cpp:220] Pointers uninitialized.
[02:01:36.3683893][INFO/main.cpp:223] Thread pool uninitialized.
[02:01:36.3683905][INFO/main.cpp:225] Farewell!

```
![ztjYrTd](https://github.com/YimMenu/YimMenu/assets/107651397/6921c3e4-c59b-4969-be2d-9525a23a572f)

### Store

Steam

### Additional context

running on my own fork build, updated to the bleeding edge. It's the base build + some still-not-merged changes for unlimited vehicle summon.
Should behave just like the latest base build, for all intents and purposes.

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by gir489returns** [comment]
#1785 should've already fixed it. My guess is the person in the passenger seat won't relinquish control to the driver. He's either running a shit menu, previous build or something else. It really should check if you're not the driver, to just give control to anyone who asks (or maybe if they're sitting in the driver's seat and they ask for security reasons.)

**Comment by GTMoraes** [comment]
> #1785 should've already fixed it. My guess is the person in the passenger seat won't relinquish control to the driver. He's either running a shit menu, previous build or something else. It really should check if you're not the driver, to just give control to anyone who asks (or maybe if they're sitting in the driver's seat and they ask for security reasons.)

I'm the one in the passenger seat, though. I'm using the latest menu.
I know the driver can sound the horn, but he can only move the vehicle when I leave the car and he lurches forward a bit. After that, everything flows as normal.

Also, other heists are still broken for me. I update the menu everyday.

**Comment by gir489returns** [comment]
I can't reproduce this. Only I was cheating, my friend was not. I was using my build of YimMenu, in both scenarios when I got inside and he got inside, we could both gain control of the vehicle. We both did the Flecca Heist with alternating control. I was host first, then he was host. You're gonna have to provide a stricter test case.

**Comment by GTMoraes** [comment]
> I can't reproduce this. Only I was cheating, my friend was not. I was using my build of YimMenu, in both scenarios when I got inside and he got inside, we could both gain control of the vehicle. We both did the Flecca Heist with alternating control. I was host first, then he was host. You're gonna have to provide a stricter test case.

I'll try reproducing it today.

Have you had issues with other missions as well?

**Comment by gir489returns** [comment]
Not since we fixed #1821 I queue up for heists quite often. I occasionally see some fagola try to cheat by teleporting me or something. For example, I was the pilot on the mission to steal the plane for the prison break, so I kicked him. But other than obvious cheaters, I have no problem with the heists, as long as someone isn't running some shit mod or something that blocks events. I've seen the prison break mission get stuck because someone was cheating.

**Comment by GTMoraes** [comment]
> Not since we fixed #1821 I queue up for heists quite often. I occasionally see some fagola try to cheat by teleporting me or something. For example, I was the pilot on the mission to steal the plane for the prison break, so I kicked him. But other than obvious cheaters, I have no problem with the heists, as long as someone isn't running some shit mod or something that blocks events. I've seen the prison break mission get stuck because someone was cheating.

That's super odd indeed.
I'll try today using my build (which is nothing more than the current build + the WIP changes on the Vehicle Controller menu), then using "vanilla" yimmenu.

**Comment by GTMoraes** [comment]
I've tried the "vanilla" YimMenu (the build from this repo) and my fork. They behave the same.

If I join an OG prep mission and it goes straight to the clothing selection after that "waiting for players" screen (where you pick the radio, if the session is open or closed etc), the mission works fine.  

But if it goes to the "select the prep mission" where I see the host picking the mission from the whiteboard, it just hangs there, without response from anyone. I can move around, click stuff, ready/unready... but nobody readys up, the host doesn't pick the mission, the Continue timer is 0:00... I've stayed there for like 5 minutes and nothing happened. 
Same thing if I join a heist ending, where the host chooses the cut for the players.
Basically, whenever the whiteboard comes, it hangs there. 

I've ran these missions using the phone option to "Join (On Call)". I've went through like 10-15 missions, and the only ones I managed to play were the ones that didn't show the whiteboard. 
I suppose there's an issue communicating with the host, or the mod refuses to receive instructions from the host on that state.

It all goes through fine if I unload YimMenu prior joining the room. I can load it after the mission begins, and all goes smooth.
